src/main/scala/tilelink/Xbar.scala

```diff
@@ -186,56 +191,105 @@ object TLXbar
 object TLXbar_ACancel
 {
   def circuit(policy: TLArbiter.Policy, seqIn: Seq[(TLBundle_ACancel, TLEdge)], seqOut: Seq[(TLBundle_ACancel, TLEdge)]): Unit = {
+    // Get [[Data]] and [[Edges]] (parameters of each port) from nodes directly.
     val (io_in, edgesIn) = seqIn.unzip
     val (io_out, edgesOut) = seqOut.unzip
 
-    // Not every master need connect to every slave on every channel; determine which connections are necessary
-    val reachableIO = edgesIn.map { cp => edgesOut.map { mp =>
-      cp.client.clients.exists { c => mp.manager.managers.exists { m =>
-        c.visibility.exists { ca => m.address.exists { ma =>
-          ca.overlaps(ma)}}}}
-      }.toVector}.toVector
-    val probeIO = (edgesIn zip reachableIO).map { case (cp, reachableO) =>
-      (edgesOut zip reachableO).map { case (mp, reachable) =>
-        reachable && cp.client.anySupportProbe && mp.manager.managers.exists(_.regionType >= RegionType.TRACKED)
-      }.toVector}.toVector
-    val releaseIO = (edgesIn zip reachableIO).map { case (cp, reachableO) =>
-      (edgesOut zip reachableO).map { case (mp, reachable) =>
-        reachable && cp.client.anySupportProbe && mp.manager.anySupportAcquireB
-      }.toVector}.toVector
-
+    // "client" is master, "manager" is slave.
+    // Not every master need connect to every slave on every channel; determine which connections are necessary. It will generate a 2 dimensional array interconnect matrix. which represents the connectability of a master port to a slave port.
+    val reachableIO = edgesIn.map {
+      // Each parameters from master.
+      cp => edgesOut.map { 
+        // Each parameters from managers (sinks).
+        mp => 
+          cp
+          // Master port parameters.
```

Comment:
These comments should not be inline in the final commit.

Reply:
has been removed